packages/web/src/components/userprofile/EditUserForm.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import {
+  Button,
+  ButtonGroup,
+  FormControl,
+  FormLabel,
+  FormHelperText,
+  Input,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalBody,
+  ModalCloseButton,
+  useDisclosure,
+} from '@chakra-ui/react';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+import { User } from '../../pages/user/[uid]';
+
+type EditFormValues = yup.InferType<typeof editFormSchema>;
+const editFormSchema = yup.object({
+  name: yup.string().trim().required('Name is required.'),
+  pronouns: yup.string(),
+  schoolName: yup.string(),
+});
+
+export interface EditUserProps {
+  setEditToggle(editToggle: boolean): void;
+  setUser(user: User): void;
+  user: {
+    id: string;
+    name: string;
+    pronouns?: string;
+    schoolName?: string;
+  };
+}
+
+export const EditUserForm: React.FC<EditUserProps> = (props: EditUserProps) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [isModalShown, setIsModalShown] = React.useState<boolean>(false);
```

Comment:
This doesn't read like it relates to an error of any sort; can you rename this so it's more clear that this is used for an error state?

Maybe even `const [errorMessage, setErrorMesage]...` and then use that to determine whether or not the modal should be shown? If you go with that approach, you can incorporate `res.text` to get valid feedback from the server to display to the user